egui/src/input_state.rs

```diff
@@ -192,6 +192,23 @@ impl InputState {
         self.pointer.wants_repaint() || self.scroll_delta != Vec2::ZERO || !self.events.is_empty()
     }
 
+    // Ignore a key if it was pressed or released this frame. Useful for hotkeys.
+    // Returns if the key was pressed this frame
+    pub fn ignore_key(&mut self, ignore_key: Key, modifiers: Modifiers) -> bool {
+        self.events = std::mem::take(&mut self.events).into_iter().filter(|event| {
+            !matches!(
+                event,
+                Event::Key {
+                    key,
+                    modifiers: mods,
+                    ..
+                } if *key == ignore_key && *mods == modifiers
```

Comment:
This will match both on press and release - I think we should only match on `pressed: true`

Reply:
Hmm, what if code depended on presses always being matched with releases? then you'd have an empty release instead of as if the event "never happened"